storage/innobase/buf/buf0buf.cc

```diff
@@ -3796,6 +3793,16 @@ void buf_pool_invalidate()
 	mysql_mutex_unlock(&buf_pool.mutex);
 }
 
+/** @return whether less than 1/4 of the buffer pool is available */
+bool buf_pool_t::running_out()
+{
+  mysql_mutex_lock(&mutex);
+  auto min_space= std::min(n_chunks, n_chunks_new) / 4 * chunks->size;
+  mysql_mutex_unlock(&mutex);
+  return !recv_recovery_is_on() &&
+    UT_LIST_GET_LEN(free) + UT_LIST_GET_LEN(LRU) < min_space;
+}
```

Comment:
The `buf_pool.mutex` is already a prominent contention point, along with `log_sys.latch`. There are rather frequent calls to `buf_pool.running_out()` for returning `DB_LOCK_TABLE_FULL` in operations that might create explicit record locks.

Can we avoid acquiring the mutex if `!resize_in_progress()`?